neovim build broken on OpenBSD

- Neovim version: master
- Operating system/version: OpenBSD 5.9-stable

### Actual behaviour
````
~/s/neovim (master|✔) ❯❯❯ gmake                     Wed Apr 20 09:37:35 PDT 2016
ninja  -C .deps
ninja: Entering directory `.deps'
[1/7] Performing build step for 'luv-static'
FAILED: cd /home/scott/src/neovim/.deps/build/src/luv-static-build && /usr/local/bin/cmake --build . && /usr/local/bin/cmake -E touch /home/scott/src/neovim/.deps/build/src/luv-static-stamp/luv-static-build
[ 50%] Building C object CMakeFiles/luv.dir/src/luv.c.o
In file included from /home/scott/src/neovim/.deps/build/src/luv/src/luv.c:41:
/home/scott/src/neovim/.deps/build/src/luv/src/dns.c: In function 'luv_getaddrinfo':
/home/scott/src/neovim/.deps/build/src/luv/src/dns.c:167: error: 'AI_V4MAPPED' undeclared (first use in this function)
/home/scott/src/neovim/.deps/build/src/luv/src/dns.c:167: error: (Each undeclared identifier is reported only once
/home/scott/src/neovim/.deps/build/src/luv/src/dns.c:167: error: for each function it appears in.)
/home/scott/src/neovim/.deps/build/src/luv/src/dns.c:171: error: 'AI_ALL' undeclared (first use in this function)
*** Error 1 in . (CMakeFiles/luv.dir/build.make:63 'CMakeFiles/luv.dir/src/luv.c.o': /usr/bin/cc   -I/home/scott/src/neovim/.deps/usr/includ...)
*** Error 1 in . (CMakeFiles/Makefile2:68 'CMakeFiles/luv.dir/all')
*** Error 1 in /home/scott/src/neovim/.deps/build/src/luv-static-build (Makefile:128 'all')
ninja: build stopped: subcommand failed.
Makefile:75: recipe for target 'deps' failed
gmake: *** [deps] Error 1
~/s/neovim (master|✔) ❯❯❯                           Wed Apr 20 09:37:49 PDT 2016
````
### Expected behaviour

Neovim builds successfully, like it used to.

### Steps to reproduce

1. `git clone --recursive https://github.com/neovim/neovim`
2. `cd neovim`
3. `echo 'CMAKE_EXTRA_FLAGS += -DENABLE_BUNDLED_JEMALLOC=OFF' > local.mk`
4. gmake

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -328,7 +328,12 @@ if((CLANG_ASAN_UBSAN OR CLANG_MSAN OR CLANG_TSAN) AND NOT CMAKE_C_COMPILER_ID MA
   message(FATAL_ERROR "Sanitizers are only supported for Clang.")
 endif()
 
-option(ENABLE_JEMALLOC "enable jemalloc" ON)
+if(CMAKE_SYSTEM_NAME STREQUAL "OpenBSD")
+  message(STATUS "detected OpenBSD; disabled jemalloc. #5318")
+  option(ENABLE_JEMALLOC "enable jemalloc" OFF)
+else()
+  option(ENABLE_JEMALLOC "enable jemalloc" ON)
+endif()
 
 if (ENABLE_JEMALLOC)
   if(CLANG_ASAN_UBSAN OR CLANG_MSAN OR CLANG_TSAN)

--- a/src/nvim/CMakeLists.txt
+++ b/src/nvim/CMakeLists.txt
@@ -120,9 +120,14 @@ if(NOT MSVC)
   set_source_files_properties(
     ${CONV_SOURCES} PROPERTIES COMPILE_FLAGS "${COMPILE_FLAGS} -Wno-conversion")
   # gperf generates ANSI-C with incorrect linkage, ignore it.
-  set_source_files_properties(
-    eval.c PROPERTIES COMPILE_FLAGS "${COMPILE_FLAGS} -Wno-static-in-inline -Wno-conversion")
-
+  check_c_compiler_flag(-Wno-static-in-inline HAS_WNO_STATIC_IN_INLINE_FLAG)
+  if(HAS_WNO_STATIC_IN_INLINE_FLAG)
+    set_source_files_properties(
+      eval.c PROPERTIES COMPILE_FLAGS "${COMPILE_FLAGS} -Wno-static-in-inline -Wno-conversion")
+  else()
+    set_source_files_properties(
+      eval.c PROPERTIES COMPILE_FLAGS "${COMPILE_FLAGS} -Wno-conversion")
+  endif()
 endif()
 
 if(DEFINED MIN_LOG_LEVEL)

